lib/elasticsearch/bulk.js

```diff
@@ -108,13 +110,33 @@ Bulk.prototype._fn = function(d) {
       });
       return items.length && items || undefined;
     },e => {
-      retry = (retry || 0);
-      if (!this.options.maxRetries || retry > this.options.maxRetries)
-        throw e;
+      retries = retries || [];
+      const retryNo = retries.length;
+      if (!this.options.maxRetries || retryNo >= this.options.maxRetries) {
+        if (e) e.retries = retries;
```

Comment:
is this just for visibility?

Reply:
Mostly for testing, i.e. showing the retries on the error object allows us to inspect.